src/main/java/racingcar/Main.java

```diff
@@ -1,17 +1,9 @@
 package racingcar;
 
-import java.util.List;
-import java.util.Random;
-import java.util.Scanner;
-
 public class Main {
     public static void main(String[] args) {
-        InputView inputView = new InputView(new Scanner(System.in));
-        int carCount = inputView.inputCarCount();
-        int tryCount = inputView.inputTryCount();
-        List<Car> cars = new CarGenerator().generateCar(carCount);
-        ResultView resultView = new ResultView();
-        RacingCarGame racingGame = new RacingCarGame(cars, tryCount, resultView);
-        racingGame.startGame(new Random());
+        RacingCarGame racingCarGame = new RacingCarGame();
+        NumberBasedCarMover carMover = new RandomNumberBasedCarMover();
+        racingCarGame.startGame(carMover);
```

Comment:
```suggestion
        NumberBasedCarMover carMover = new RandomNumberBasedCarMover();
        TryCount tryCount = new TryCount(InputView.inputTryCount());
        Cars cars = new Cars(InputView.inputCarNames());

        RacingCarGame racingCarGame = new RacingCarGame(cars, tryCount);
        racingCarGame.startGame(carMover);
```

객체를 만들었다면, 상태를 관리할수있는 객체로 만들면 어떨까요?
어떻게 생각하시나요? 😄 

더불어, 테스트에 좀 더 용이하지 않을까요?

Reply:
상태를 관리할수있는 객체 라는게 와닿질 않는데 어떻게 바꾸면 좋을까요?

Reply:
```
public class RacingCarGame {
    private TryCount tryCount;
    private Cars cars;
    private NumberBasedCarMover carMover;

    public RacingCarGame(TryCount tryCount, Cars cars) {
        this.tryCount = tryCount;
        this.cars = cars;
    }
```

---

현재 `RacingCarGame` 클래스에서 의존성을 생성자 주입 방식으로 처리하여, 내부 상태를 외부에서 제어할 수 있도록 한 점이 테스트 용이성 측면에서 긍정적이라고 생각해요.

예를 들어, 아래의 `startGame` 메소드처럼 내부에서 `Cars`와 `TryCount`를 직접 생성한다면 외부에서 제어할 수 없어 테스트하기 어려워집니다.

```java
public void startGame(NumberBasedCarMover carMover) {
    Cars cars = inputCars();
    TryCount tryCount = inputTryCount();
    game(cars, tryCount, carMover);
    printGameResult(cars);
}
```

이 경우, 객체 내부에 상태가 고정되어 있기 때문에 테스트 시에 초기 상태나 외부 인자들을 주입할 방법이 제한됩니다.  
따라서, 초기 상태나 변경 가능한 인자들을 외부에서 주입받도록 구조를 개선한다면, `startGame`과 같은 메소드의 테스트가 훨씬 수월해질 것이라고 생각합니다.

---

이렇게 개선하면 객체의 상태를 외부에서 관리할 수 있어 단위 테스트 뿐만 아니라 전체적인 코드 유지보수성도 좋아질것 같아서요~ 😄